connector/spanmetricsconnectorv2/config.go

```diff
@@ -0,0 +1,165 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package spanmetricsconnectorv2 // import "github.com/elastic/opentelemetry-collector-components/connector/spanmetricsconnectorv2"
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+const (
+	defaultMetricNameSpans = "trace.span.duration"
+	defaultMetricDescSpans = "Observed span duration."
+)
+
+var (
+	defaultHistogramBucketsMs = [...]float64{2, 4, 6, 8, 10, 50, 100, 200, 400, 800, 1000, 1400, 2000, 5000, 10_000, 15_000}
+)
+
+type MetricUnit string
+
+const (
+	MetricUnitMs MetricUnit = "ms"
+	MetricUnitS  MetricUnit = "s"
+)
+
+// Config for the connector
+type Config struct {
+	Spans map[string]MetricInfo `mapstructure:"spans"`
+}
+
+// MetricInfo for a data type
+type MetricInfo struct {
+	Description string            `mapstructure:"description"`
+	Attributes  []AttributeConfig `mapstructure:"attributes"`
```

Comment:
The `countconnector` calls it `attributes` and it made more sense to me at the time of writing since it relates to the OTel data model. Though, it might be prudent to keep it dimension if we are planning on upstreaming this. What was your reasoning behind using dimensions?

Reply:
Just to minimise UX differences between spanmetricsconnector and spanmetricsconnectorv2. But if `attributes` is in use in the `countconnector`, I can also buy consolidating on one and switching to `attributes`.

Reply:
If the proposal gets accepted to replace the current span metrics connector then it would be a breaking change anyway (producing aggregated metrics vs non-aggregated ones) so we could take the opportunity to consolidate the configuration. If required, we can provide a form of aliasing of dimensions -> attributes for compatibility.

Reply:
Fair enough. Switching to `attributes` sounds fine to me.